More docs on migration CLI (#2279)
edgedb/edgedb: docs/cli/edgedb_migration.rst

multiple ``.esdl`` files. The migration tools will read all of them
and treat them as a single SDL document.

> Here is typical file tree::

Reviewer: ```suggestion
Here is a typical file tree::
```

>     ❯ tree dbschema

Reviewer 2: Inconsistent use of shell prefixes, maybe stick to `$` since they are used everywhere else in documentation?

Author: Fixed

    dbschema
    ├── default.esdl
    └── migrations
        ├── 00001.edgeql
        ├── 00002.edgeql
        ├── 00003.edgeql
        ├── ....
        └── 00019.edgeql

In the tree:

> * ``dbschema/*.esdl`` is a set of files that keep current database schema

Reviewer 2: is a set of files that ~~keep~~ contains current database schema

> * ``dbschema/*.esdl`` is a set of files that contains current database schema

Reviewer: ```suggestion
* ``dbschema/*.esdl`` is a set of files that describes the current database schema
```

> * ``default.esdl`` is named after ``module default`` that is usually used for

Reviewer: ```suggestion
* ``default.esdl`` is named after ``module default``, which is normally used for
```

>   initial application schema. You don't have to stick to single file per

Reviewer: ```suggestion
  the initial application schema. There is no requirement to use a single
```

>   module, but this is a good start.

Reviewer: ```suggestion
  file per module, but it is a recommended style.
```

Author: I'm not entirely sure that this is a "recommeded style". I had a temptation to split schema into multiple files already at few hundreds of lines of schema. But I still don't know at which size of schema I would create modules.

> * ``dbschema/migrations/*.edgeql`` is a sequence of migration files that when

Reviewer: ```suggestion
* ``dbschema/migrations/*.edgeql`` is a sequence of migration files that, when
```

>   applied bring database schema to the current. These files are mostly managed

Reviewer 2: > bring database schema to the current.

bring database schema to the latest version?

Author: Fixed

>   applied bring database schema to the latest version. These files are mostly

Reviewer: ```suggestion
  applied, bring the database schema to the latest version. These files are mostly
```

>   managed by command-line tools, but have to be commited to source code

Reviewer: ```suggestion
  managed by the command-line tools, and normally are committed to the source
```

>   repository, and occasionally may need to be edited manually (common example

Reviewer: ```suggestion
  repository, and may occasionally need to be edited manually (a common example
```

  branch).

.. _ref_cli_edgedb_migration_workflow:

General Workflow
================

1. Create or edit schema in ``dbschema/*.esdl`` files
2. Before applying schema, create a migration file:

   .. code-block:: bash

      edgedb create-migration

   It runs through changes interactively and puts
   ``dbschema/migrations/<revision-number>.edgeql`` file into the file system
   (later you want to check it into a version control)

> 3. To apply schema run:

Reviewer 2: Maybe simplify this list?
```
1. Create or edit schema in ``dbschema/*.esdl`` files
2. Create migration file
3. Apply migration
```

Author: This is a quick refresher, if you didn't create any migrations for some time. Without commands, you have to jump around sections to figure out what you are actually supposed to do.

   .. code-block:: bash

      edgedb migrate

> It's worth mentioning that after code checkout (e.g. ``git pull``) it makes

Reviewer: ```suggestion
It's worth mentioning that after a code checkout (e.g. ``git pull``) it makes
```

> sense to run ``edgedb migrate`` again. Server has the latest migration id

Reviewer: ```suggestion
sense to run ``edgedb migrate`` again. The server has a record of the latest
```

> recorded so will only apply new migrations.

Reviewer: ```suggestion
migration, so it will only apply new migrations.
```

> At any time ``edgedb show-status`` describes if there are any pending

Reviewer: ```suggestion
At any time ``edgedb show-status`` outputs a list of pending
```

Author: It doesn't output a list of pending schema changes, we don't have a way to do that yet. And it doesn't output a list off pending migrations too (what if there are 100?). It only shows status, which is a number of persnding migrations and first change that is going to be inserted with create migrations. But I don't think we want to describe it's output at this precision.

migrations or schema changes.

.. _ref_cli_edgedb_create_migration:

Create migration script
=======================

> After editing schema files, migration script must be created.

Reviewer: ```suggestion
After editing schema files, a migration script needs to be created.
```

> Each time you've edit schema, the next step is creating a migration script.

Reviewer 2: After editing schema files, migration script must be created.

``dbschema/migrations/`` directory and sequentially run them on the
target instance.

.. _ref_cli_edgedb_migrate:

Show Status
===========

To figure out the status of the schema and migrations use the respective
command:

.. cli:synopsis::

>     edgedb -I <instance-name> show-status [migration-option...]

Reviewer 2: Why does this part mention `-I <instance-name>` while previous doesn't? `edgedb migrate` below doesn't mention it too

Author: It's mentioned in synopsis lines, where we're trying to give comprehensive guide on the command. But in summary, it's too much noise. It's also not strictly required, you can also specify `--host=` or `--dsn=` instead.

And we are going to drop ubiquitous usage of `-I` with project-local config in a month or so, so I wouldn't concentrate on that too much.
